pkg/ingester/ingester_v2.go

```diff
@@ -696,7 +719,14 @@ func (i *Ingester) v2Push(ctx context.Context, req *cortexpb.WriteRequest) (*cor
 		return nil, err
 	}
 
-	db, err := i.getOrCreateTSDB(userID, false)
+	gl := i.getGlobalLimits()
+	if gl != nil && gl.MaxIngestionRate > 0 {
+		if rate := i.ingestionRate.rate(); rate >= gl.MaxIngestionRate {
```

Comment:
`.rate()` takes an lock. I'm a bit worried about it. Should we change it to a RWLock?

Reply:
Good point.